pkg/cloud/vsphere/machineactuator.go

```diff
@@ -54,22 +64,116 @@ func NewGovmomiMachineActuator(m manager.Manager, clusterV1alpha1 clusterv1alpha
 
 	return &VsphereClient{
 		clusterV1alpha1:  clusterV1alpha1,
+		lister:           lister,
 		controllerClient: m.GetClient(),
 		provisioner:      provisioner,
 	}, nil
 }
 
-func (vc *VsphereClient) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
-	if vc.provisioner != nil {
-		err := vc.provisioner.Create(ctx, cluster, machine)
-		if err != nil {
-			klog.Error(err)
-			return err
+func (vc *VsphereClient) Create(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+	machine *clusterv1.Machine) error {
+
+	if vc.provisioner == nil {
+		return errors.New("No provisioner available")
+	}
+
+	if cluster == nil {
+		return errors.Errorf(
+			"missing cluster for machine %s/%s",
+			machine.Namespace, machine.Name)
+	}
+
+	machineRole := vsphereutils.GetMachineRole(machine)
+	if machineRole == "" {
+		return errors.Errorf(
+			"Unable to get machine role while creating machine "+
+				"%s=%s %s=%s %s=%s %s=%s",
+			"cluster-namespace", cluster.Namespace,
+			"cluster-name", cluster.Namespace,
+			"machine-namespace", machine.Namespace,
+			"machine-name", machine.Name)
+	}
+
+	klog.V(2).Infof("Creating machine in cluster %s=%s %s=%s %s=%s %s=%s %s=%s",
+		"cluster-namespace", cluster.Namespace,
+		"cluster-name", cluster.Namespace,
+		"machine-namespace", machine.Namespace,
+		"machine-name", machine.Name,
+		"machine-role", machineRole)
+
+	controlPlaneMachines, err := vsphereutils.GetControlPlaneMachinesForCluster(cluster, vc.lister)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"unable to get control plane machines while creating machine "+
+				"%s=%s %s=%s %s=%s %s=%s",
+			"cluster-namespace", cluster.Namespace,
+			"cluster-name", cluster.Namespace,
+			"machine-namespace", machine.Namespace,
+			"machine-name", machine.Name)
+	}
+
+	// Init the control plane by creating this machine.
+	if machineRole == "controlplane" && len(controlPlaneMachines) <= 1 {
```

Comment:
Should `controlplane` be a const?

Reply:
Also `len(controlPlaneMachines) == 0`?

Reply:
Re: constant, probably, but it's not used in too many places. I'll update it regardless.